src/vopr/main.go

```diff
@@ -0,0 +1,766 @@
+
+	// Extract the information about the conditions under which the Vopr runs TigerBeetle.
+	output.extract_parameters(&message)
+
+	length_of_stack_trace := len(output.stack_trace)
+	length_of_parameters := len(output.parameters)
+	length_of_logs := len(output.logs)
+	remaining_space := MAX_GITHUB_ISSUE_SIZE
+
+	stack_trace := ""
+	parameters := ""
+	debug_logs := ""
+
+	// Set the stack trace string
+	if length_of_stack_trace > 0 {
+		if length_of_stack_trace <= MAX_GITHUB_ISSUE_SIZE {
+			stack_trace = make_markdown_compatible(string(output.stack_trace[:]))
+			remaining_space -= length_of_stack_trace
+		} else {
+			// If the stack trace is too large then just capture the beginning of it.
+			stack_trace = make_markdown_compatible(string(output.stack_trace[:MAX_GITHUB_ISSUE_SIZE-4]))
+			stack_trace += "..."
+			remaining_space = 0
+		}
+	}
+
+	// Set the parameters string
+	if length_of_parameters > 0 && remaining_space >= min_useful_length {
+		if length_of_parameters <= remaining_space {
+			parameters = make_markdown_compatible(output.parameters[:])
+			remaining_space -= length_of_parameters
+		} else {
+			// If the parameters section is too large then just capture the beginning of it.
+			parameters = make_markdown_compatible(output.parameters[:remaining_space-4]) + "..."
+			remaining_space = 0
+		}
+	}
+
+	// Set the debug logs string
+	if remaining_space >= min_useful_length {
+		if length_of_logs < remaining_space {
+			debug_logs = make_markdown_compatible(string(output.logs[:]))
+		} else {
+			// Get the tail of the logs
+			tail_start_index := length_of_logs - remaining_space
+			debug_logs = make_markdown_compatible(string(output.logs[tail_start_index:]))
+		}
+	}
+
+	var bugType string
+	if message.bug == 1 {
+		bugType = "correctness"
+	} else if message.bug == 2 {
+		bugType = "liveness"
+	} else if message.bug == 3 {
+		bugType = "crash"
+	}
+
+	var body string
+	if len(parameters) > 0 {
+		body += fmt.Sprintf(
+			"**Parameters:**<br>%s<br><br>",
+			parameters,
+		)
+	}
+	if len(stack_trace) > 0 {
+		body += fmt.Sprintf(
+			"**Stack Trace:**<br>%s<br><br>",
+			stack_trace,
+		)
+	}
+	if len(debug_logs) > 0 {
+		body += fmt.Sprintf(
+			"**Tail of Debug Logs:**<br>%s<br><br>",
+			debug_logs,
+		)
+	}
+
+	timestamp := time.Now().UTC().String()
+
+	markdown := fmt.Sprintf(
+		"**Bug Type:**<br>%s<br><br>**Seed:**<br>%d<br><br>**Commit Hash:**<br>%s<br><br>%s**Time:**<br>%s",
+		bugType,
+		message.seed,
+		hex.EncodeToString(message.commit[:]),
+		body,
+		timestamp,
+	)
+	return markdown
+}
+
+// Escape characters that have special use in Markdown.
+func make_markdown_compatible(text string) string {
+	text = strings.ReplaceAll(text, "\"", "\\\"")
+	text = strings.ReplaceAll(text, "<", "&lt;")
+	text = strings.ReplaceAll(text, ">", "&gt;")
+	text = strings.ReplaceAll(text, "\n", "<br>")
+	return text
+}
+
+// Retrieve all the required environmental variables up front.
+func set_environmental_variables() {
```

Comment:
```suggestion
// Retrieve all the required environment variables up front.
func set_environment_variables() {
```

To match how they are typically called "environment variables".